datafusion/common/src/config.rs

```diff
@@ -212,11 +219,12 @@ config_namespace! {
         pub enable_ident_normalization: bool, default = true
 
         /// When set to true, SQL parser will normalize options value (convert value to lowercase)
-        pub enable_options_value_normalization: bool, default = true
+        pub enable_options_value_normalization: bool, default = false
```

Comment:
This is the important change in this PR, along with the definition of `transform` for option values.

cc @alamb @comphead @jayzhan211

Reply:
FWI @xinlifoobar and @berkaysynnada  who worked on https://github.com/apache/datafusion/pull/11330 where this feature was added. Interestingly, this option was added in response to seemingly the same problem (AWS access keys being normalized) as faced by @tinfoil-knight here:
- https://github.com/apache/datafusion/issues/10853


Rather than changing the default config values, we could potentially change the default in datafusion cli and/or improve documentation as well

Reply:
Looking at the linked issue and its PR, it seems what I am looking for is exactly what @ozankabak wanted in https://github.com/apache/datafusion/pull/11330#issuecomment-2246242461

> We would like to have key-level control on what normalization means for config options, built-in and extension.

this PR addresses this for built-in options. A nice generic normalization for extension options is yet to be addressed.